src/components/refactor/dialogs/lines-attach-to-split-lines/lines-attach-to-split-lines-form.js

```diff
@@ -0,0 +1,229 @@
+        name: `${VOLTAGE_LEVEL_ID}`,
+    });
+
+    const getObjectId = useCallback((object) => {
+        if (typeof object === 'string') {
+            return object;
+        }
+
+        return object?.id;
+    }, []);
+
+    const resetBusBarSection = useCallback(() => {
+        setValue(`${BUS_BAR_SECTION_ID}`, null);
+    }, [setValue]);
+
+    useEffect(() => {
+        if (studyUuid && currentNodeUuid)
+            fetchVoltageLevelsIdAndTopology(studyUuid, currentNodeUuid).then(
+                (values) => {
+                    setVoltageLevelOptions(
+                        values.sort((a, b) => a.id.localeCompare(b.id))
+                    );
+                }
+            );
+    }, [studyUuid, currentNodeUuid]);
+
+    useEffect(() => {
+        fetchEquipmentsIds(
+            studyUuid,
+            currentNodeUuid,
+            undefined,
+            'LINE',
+            true
+        ).then((values) => {
+            setLinesIds(values);
+        });
+    }, [studyUuid, currentNodeUuid]);
+
+    useEffect(() => {
+        if (watchVoltageLevelId) {
+            const voltageLevel = voltageLevelOptions.find(
+                (voltageLevel) => voltageLevel[ID] === watchVoltageLevelId
+            );
+            const topologyKind = voltageLevel?.[TOPOLOGY_KIND];
+
+            switch (topologyKind) {
+                case 'NODE_BREAKER':
+                    fetchBusbarSectionsForVoltageLevel(
+                        studyUuid,
+                        currentNodeUuid,
+                        watchVoltageLevelId
+                    ).then((busbarSections) => {
+                        const busbarSectionsIds = busbarSections?.map(
+                            (busbarSection) => busbarSection[ID]
+                        );
+                        setBusOrBusbarSectionOptions(busbarSectionsIds);
+                    });
+                    break;
+
+                case 'BUS_BREAKER':
+                    fetchBusesForVoltageLevel(
+                        studyUuid,
+                        currentNodeUuid,
+                        watchVoltageLevelId
+                    ).then((buses) => {
+                        const busesIds = buses?.map((buse) => buse[ID]);
+                        setBusOrBusbarSectionOptions(busesIds);
+                    });
+                    break;
+
+                default:
+                    setBusOrBusbarSectionOptions([]);
+                    break;
+            }
+        } else {
+            setBusOrBusbarSectionOptions([]);
+        }
+    }, [watchVoltageLevelId, studyUuid, currentNodeUuid, voltageLevelOptions]);
+
+    const lineToAttachTo1Field = (
+        <AutocompleteInput
+            outputTransform={(value) => (value === '' ? null : value)}
+            allowNewValue
+            forcePopupIcon
+            name={LINE_TO_ATTACH_TO_ID_1}
+            label="Line1"
+            options={linesIds?.sort((a, b) => a.localeCompare(b))}
```

Comment:
make a usememo of "linesIds?.sort((a, b) => a.localeCompare(b))"
here we are sorting the array for each field, which can be very expensive when loading a lot of lines